src/plugins/discover/public/application/components/doc/use_es_doc_search.ts

```diff
@@ -52,6 +46,9 @@ export function buildSearchBody(
     // @ts-expect-error
     request.body.fields = [{ field: '*', include_unmapped: 'true' }];
     request.body.runtime_mappings = runtimeFields ? runtimeFields : {};
+    if (requestAllFields) {
```

Comment:
Suppose whould be better to use `isJsonTab`, since `_source` should be retrived only when navigating JSON tab as Tim described [**here**](https://github.com/elastic/kibana/issues/95946#issuecomment-852911748). As I understand this is the reason why this check is exists here, because we can omit it and fetch`_source` 
 on clicking `Table` tab too. (https://github.com/elastic/kibana/issues/95946#issuecomment-852911748)